releasenotes-builder/src/test/java/com/github/checkstyle/internal/AbstractPathTestSupport.java

```diff
@@ -0,0 +1,124 @@
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.github.checkstyle.internal;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
+
+import com.github.checkstyle.globals.Constants;
+
+public abstract class AbstractPathTestSupport {
+
+    protected static final String BUG = Constants.BUG_LABEL;
+    protected static final String NEW_FEATURE = Constants.NEW_FEATURE_LABEL;
+    protected static final String NEW_MODULE = Constants.NEW_MODULE_LABEL;
+    protected static final String MISC = Constants.MISCELLANEOUS_LABEL;
+    protected static final String BREAKING = Constants.BREAKING_COMPATIBILITY_LABEL;
+
+    @Rule
+    public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Before
+    public void setup() throws IOException {
+        temporaryFolder.delete();
+        temporaryFolder.create();
+    }
+
+    /**
+     * Retrieves the temporary folder.
+     *
+     * @return The temporary folder.
+     */
+    protected File getTempFolder() {
+        return temporaryFolder.getRoot();
+    }
+
+    /**
+     * Returns canonical path for the file with the given file name.
+     * The path is formed base on the root location.
+     * This implementation uses 'src/test/resources/'
+     * as a root location.
+     *
+     * @param fileName file name.
+     * @return canonical path for the file name.
+     * @throws IOException if I/O exception occurs while forming the path.
+     */
+    protected static String getPath(String fileName) throws IOException {
+        return new File("src/test/resources/com/github/checkstyle/" + fileName).getCanonicalPath();
+    }
+
+    /**
+     * Asserts file equals the expected result.
+     *
+     * @param expectedName Expected file path to read.
+     * @param actualName Actual file path to read.
+     * @throws IOException if there is an error reading the file.
+     */
+    protected void assertFile(String expectedName, String actualName) throws IOException {
+        final String expected =
+                getFileContents(new File(getPath(expectedName)));
+        final String actual = getFileContents(new File(getTempFolder(), actualName))
+                .replace(LocalDate.now()
+                                .format(DateTimeFormatter.ofPattern("dd.MM.yyyy", Locale.US)),
+                        "XX.XX.XXXX");
+
+        Assert.assertEquals("expected " + actualName, expected, actual);
```

Comment:
The code in this method was moved, but it was also changed because it was specific to xdoc and used for more file. Also changed the assertion to output the file being checked.

Reply:
Agreed. Would make life easier to see the actual file.